crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs

```diff
@@ -122,70 +122,75 @@ impl GeneratedAcir {
     pub(crate) fn call_black_box(
         &mut self,
         func_name: BlackBoxFunc,
-        mut inputs: Vec<FunctionInput>,
+        inputs: &[Vec<FunctionInput>],
         constants: Vec<FieldElement>,
+        output_count: usize,
     ) -> Result<Vec<Witness>, InternalError> {
-        intrinsics_check_inputs(func_name, &inputs)?;
+        let input_count = inputs.iter().fold(0usize, |sum, val| sum + val.len());
+        intrinsics_check_inputs(func_name, input_count);
+        intrinsics_check_outputs(func_name, output_count);
 
-        let output_count = black_box_expected_output_size(func_name)?;
         let outputs = vecmap(0..output_count, |_| self.next_witness_index());
 
         // clone is needed since outputs is moved when used in blackbox function.
         let outputs_clone = outputs.clone();
 
         let black_box_func_call = match func_name {
             BlackBoxFunc::AND => {
-                BlackBoxFuncCall::AND { lhs: inputs[0], rhs: inputs[1], output: outputs[0] }
+                BlackBoxFuncCall::AND { lhs: inputs[0][0], rhs: inputs[1][0], output: outputs[0] }
```

Comment:
the [0][0] could maybe use an extracted variable or a type so it's not a 2d array

Reply:
Going to leave like this for now and we can consider a separate enum on black box func inputs that differentiates between single values and arrays. This may be more complex though for only having two cases